src/solvers/strings/string_constraint_generator_main.cpp

```diff
@@ -114,6 +114,19 @@ void merge(string_constraintst &result, string_constraintst other)
     std::back_inserter(result.not_contains));
 }
 
+and_exprt char_set_restrictions(const exprt &expr, const std::string &char_set)
+{
+  // Parse char_set
+  PRECONDITION(char_set.length() == 3);
+  PRECONDITION(char_set[1] == '-');
+  const char &low_char = char_set[0];
+  const char &high_char = char_set[2];
+
+  return and_exprt(
+    binary_relation_exprt(expr, ID_ge, from_integer(low_char, expr.type())),
+    binary_relation_exprt(expr, ID_le, from_integer(high_char, expr.type())));
```

Comment:
💡 What helped me parse these things in the past - a formula in a comment above the block, like so:
```
// expr >= low_char && expr <= high_char
```

Reply:
Are you feeling OK?